File: frontend/src/types/session.ts
export interface ChatSession {
  id: string
  title: string
  query_mode: string
  message_count: number
  last_message_at: string | null
  is_compacted: boolean
  created_at: string
}

export interface SessionListResponse {
  sessions: ChatSession[]
  total: number
}

export interface SessionCreateRequest {
  query_mode?: string
}

export interface SessionRenameRequest {
  title: string
}

File: backend/app/schemas/session.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class SessionCreateRequest(BaseModel):
    query_mode: str = "hybrid"


class SessionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    query_mode: str
    message_count: int
    last_message_at: datetime | None
    is_compacted: bool
    created_at: datetime


class SessionListResponse(BaseModel):
    sessions: list[SessionResponse]
    total: int


class SessionRenameRequest(BaseModel):
    title: str
